backend/shelf-server/src/main/java/com/htec/shelfserver/controller/UserController.java

```diff
@@ -29,11 +30,21 @@ public ResponseEntity getUserById(@AuthenticationUser AuthUser user, @PathVariab
     }
 
     @PostMapping(value = "/register", produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<TextResponseMessage> createUser(@RequestBody UserRequestModel userRequestModel) {
+    public ResponseEntity<TextResponseMessage> registerUser(@RequestBody UserRegisterRequestModel userRegisterRequestModel) {
 
-        UserDTO userDTO = UserMapper.INSTANCE.userRequestModelToUserDto(userRequestModel);
+        UserDTO userDTO = UserMapper.INSTANCE.userRegisterRequestModelToUserDto(userRegisterRequestModel);
 
-        userService.createUser(userDTO);
+        userService.registerUser(userDTO);
+
+        return ResponseEntity.status(HttpStatus.CREATED).body(new TextResponseMessage("User registered", HttpStatus.CREATED.value()));
+    }
+
+    @PostMapping(value = "/register/microsoft", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<TextResponseMessage> registerUserMicrosoft(@RequestBody UserRegisterMicrosoftRequestModel userRegisterMicrosoftRequestModel) {
+
+        UserDTO userDTO = UserMapper.INSTANCE.userRegisterMicrosoftRequestModelToUserDto(userRegisterMicrosoftRequestModel);
```

Comment:
What is here going to be mapped?

Reply:
bearerToken only

Reply:
I think that this is not needed, cause you need bearerToken to call Microsoft for informations which then you can map to UserDTO, or think to do it differently ?